### DISCUSSION TITLE
redisInitiateSSLWithContext() API failing when upgraded Hiredis code to use SSL Context

### DISCUSSION THREAD

**Discussion by nitinjm123** [redis/hiredis#1037] [Q&A]
Hi,

I have updated Hiredis code to use SSL connection to existing working code.
As per example, followed the way to create RedisSSLContext and called redisInitiateSSLWithContext() API.

#1. redisInitOpenSSL();

#2. redisSSLContext* pRedisSSLContext = redisCreateSSLContext(
			NULL,     /* File name of trusted CA/ca bundle file, optional */
			NULL,       /* Path of trusted certificates, optional */
			NULL,      /* File name of client certificate file, optional */
			NULL,       /* File name of client private key, optional */
			server,	 /* Server name to request (SNI), optional */
			&SSLContextError);

#3. created redisContext* pRedisContext with redisConnectWithTimeout()
    This is successful.
	
$4. called redisInitiateSSLWithContext(pRedisContext, pRedisSSLContext)
     Result is not successful (!REDIS_OK)
	 pRedisContext->err =1, 
	 ptHandle->pRedisContext->errstr=Unknown error
	 
Can someone guide me to overcome this issue and find the exact cause?

Regards,
Nitin

**Comment by michael-grunder** [comment]
It's hard to say without a small example I can compile and run.

Are you able to connect to the Redis instance via `redis-cli`?  Since it uses hiredis to connect you may be able to get a hint as to what it is doing differently.

Since you aren't attaching any certs or keys you will only be able to connect to a `redis-server` that was started with the `--insecure` option.

**Reply by nitinjm123** [reply]
Thanks Michael.
Yes, I am able to connect to Redis instance via redis-cli .

From program, AUTH command failed for 6380 SSL.

Query: Does redis client support ssl connection without certificate?

**Reply by michael-grunder** [reply]
Yes, but whether or not `redis-server` will accept such a connection depends on how it was started.  If you have access to the Redis server logs you can see if the `redis-cli` connection is being rejected due to a TLS error.

**Reply by nitinjm123** [reply]
> Yes, but whether or not `redis-server` will accept such a connection depends on how it was started. If you have access to the Redis server logs you can see if the `redis-cli` connection is being rejected due to a TLS error.

Through redis-cli both NonSSL and SSL connections are successful:
Non-SSL connection:
rdcli -h xxx.redis.xxx.xxx.net -a passwordString -p 6379

SSL-Connection:
rdcli --tls -h xxx.redis.xxx.xxx.net -a passwordString -p 6380
